design/routecrd.md

```diff
@@ -0,0 +1,88 @@
+# Route CRD
+
+## Executive Summary
+
+The current Kubernetes ingress spec does not fully represent a real-world ingress object. There are many details which cannot be described without using annotations. For example, features such as automatic retries, active health checking, circuit breaking, global rate limiting, request shadowing, and zone-local load balancing are items that are missing clear representation in the current specification. 
+
+A way to address these requirements are to utilize Custom Resource Definitions, allowing for clear representation and validation of those objects. A Route CRD will be created which will define the ingress specification and provide the details to route incoming traffic via a host, apply corresponding pathPrefixes and finally define the upstream endpoints who will serve the request. 
+
+The Route CRD defines the Ingress route for a single vhosts routing to one or more upstream services. The Route CRD allows users also define a port to proxy as well as define a health check endpoint. 
+
+## Goals
+
+- Enable ingress specification via Kubernetes Custom Resource Definition
+- Provide field level validation on CRD
+
+## Terminology / Spec
+
+Following is an layout of the spec which the CRD will be implemented:
+
+- **apiVersion**: contour.heptio.com/v1alpha
+- **kind**: Route
+- **metadata**:
+  - **namespace**: Namespace deployed into
+  - **name**: Name of object
+- **status**: 
+  - **currentStatus**: Current Status of the route (e.g. Invalid, Error, etc)
+  - **lastProcessTime**: Time the object was processed by Contour and made available to cluster
+  - **[]errors**: List of errors found in configuration
+- **spec**:
+  - **strategy (Optional)**: Load balancer algorithm type: Round Robin / Weighted Least Request / Random (Note: A strategy defined here will be the default for all routes below unless overridden in route spec)
```

Comment:
Seems like the strategy should be a required field. Unless we have a default strategy that would apply if the strategy is not specified here nor in the routes.

Reply:
I added a note that it defaults to `RoundRobin` if not specified.